onnxruntime/test/fuzzing/src/test.cpp

```diff
@@ -285,123 +280,113 @@ int main(int argc, char* argv[])
-        Logger::testLog << L"Number of unique errors: "<< werrStreamBuf.get_unique_errors() << L"\n";
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.disable();
-          Logger::testLog.flush();
+      } else {
+        throw std::exception("Unable to initialize the Onnx model in memory");
+      }
+    } else {
+      std::wstring ortModelFile = modelFile + L".ort";
+      Ort::SessionOptions so;
+      Ort::Env env;
+      so.SetGraphOptimizationLevel(ORT_DISABLE_ALL);
+      so.SetOptimizedModelFilePath(ortModelFile.c_str());
+      so.AddConfigEntry(kOrtSessionOptionsConfigSaveModelFormat, "ORT");
+      Ort::Session session(env, modelFile.c_str(), so);
+      size_t numBytes = std::filesystem::file_size(ortModelFile);
+      std::vector<char> modelData(numBytes);
+      std::ifstream ortModelStream(ortModelFile, std::ifstream::in | std::ifstream::binary);
+      ortModelStream.read(modelData.data(), numBytes);
+      ortModelStream.close();
+      for (int i = 8; i < numBytes; i++) {
+        char tmp = modelData[i];
+        int j = (numBytes - i - 1) ? (i + 1) : 0;  // j = (i+1) % (numBytes -1)
+        modelData[i] ^= modelData[j];
+        modelData[i] = tmp;
+        try {
+          Logger::testLog << "Starting Test" << Logger::endl;
+          OnnxPrediction predict(modelData);
+          predict.SetupInput(GenerateDataForInputTypeTensor, 0);
+          predict.RunInference();
+          predict.PrintOutputValues();
+          num_successful_runs++;
+          Logger::testLog << "Completed Test iteration: " << i << Logger::endl;
+        } catch (...) {
+          fuzz_handle_exception();
         }
       }
     }
-    else
-    {
-      throw std::exception("Unable to initialize the Onnx model in memory");
+    Logger::testLog << "Ending Test" << Logger::endl;
+
+    if (userOpt.stress) {
+      Logger::testLog.enable();
     }
 
+    Logger::testLog << L"Total number of exceptions: " << num_unknown_exception + num_std_exception + num_ort_exception << Logger::endl;
+    Logger::testLog << L"Number of Unknown exceptions: " << num_unknown_exception << Logger::endl;
+    Logger::testLog << L"Number of ort exceptions: " << num_ort_exception << Logger::endl;
+    Logger::testLog << L"Number of std exceptions: " << num_std_exception << Logger::endl;
+    Logger::testLog << L"Number of unique errors: " << werrStreamBuf.get_unique_errors() << L"\n";
+
+    if (userOpt.stress) {
+      Logger::testLog.disable();
+      Logger::testLog.flush();
+    }
     return 0;
-  }
-  catch(const Ort::Exception& ortException)
-  {
-    Logger::testLog<< L"onnx runtime exception: " << ortException.what() << Logger::endl;
-  }
-  catch(const std::exception& e)
-  {
-    Logger::testLog<< L"standard exception: " << e.what() << Logger::endl;
-  }
-  catch(...)
-  {
-    Logger::testLog<< L"Something Went very wrong: " << Logger::endl;
+  } catch (const Ort::Exception& ortException) {
```

Comment:
If there's a non-ORT error should this return a non-zero value?

Reply:
Sorry - the diffs make it a bit hard to read without using an external diff too. Looks like it's similar to the old logic, but not quite following why we have a catch for an OrtException outside of the test loop. What is actually being tested and expected is unclear. I would have expected there was a section of code with all the ORT calls that is tested. Within that section an OrtException is fine, a std::exception is probably ok but not expected, and anything else is bad. In the main program, I would have thought any exception was bad and coming from the early exit path in the test code where it rethrows in the catch-all.

---
In reply to: [521695691](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521695691) [](ancestors = 521695691)

Reply:
Broadly speaking the try-catch block in inner loop is to catch the onnxruntime related exceptions and outer try-catch block is catch exceptions such as file-not-found, permissions, etc.

---
In reply to: [521718084](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521718084) [](ancestors = 521718084,521695691)